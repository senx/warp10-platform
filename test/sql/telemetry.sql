-- This file and its contents are licensed under the Apache License 2.0.
-- Please see the included NOTICE for copyright information and
-- LICENSE-APACHE for a copy of the license.

\c :TEST_DBNAME :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status(int) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status_ssl(int) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status_ssl' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status_mock(text) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status_mock' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_validate_server_version(response text)
    RETURNS TEXT
    AS :MODULE_PATHNAME, 'ts_test_validate_server_version' LANGUAGE C IMMUTABLE PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_telemetry_main_conn(text, text)
RETURNS BOOLEAN AS :MODULE_PATHNAME, 'ts_test_telemetry_main_conn' LANGUAGE C IMMUTABLE PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_telemetry(host text = NULL, servname text = NULL, port int = NULL) RETURNS JSONB AS :MODULE_PATHNAME, 'ts_test_telemetry' LANGUAGE C IMMUTABLE PARALLEL SAFE;

INSERT INTO _timescaledb_catalog.metadata VALUES ('foo','bar',TRUE);
INSERT INTO _timescaledb_catalog.metadata VALUES ('bar','baz',FALSE);

\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
SELECT _timescaledb_internal.test_status_ssl(200);
SELECT _timescaledb_internal.test_status_ssl(201);
\set ON_ERROR_STOP 0
SELECT _timescaledb_internal.test_status_ssl(304);
SELECT _timescaledb_internal.test_status_ssl(400);
SELECT _timescaledb_internal.test_status_ssl(401);
SELECT _timescaledb_internal.test_status_ssl(404);
SELECT _timescaledb_internal.test_status_ssl(500);
SELECT _timescaledb_internal.test_status_ssl(503);
\set ON_ERROR_STOP 1
SELECT _timescaledb_internal.test_status(200);
SELECT _timescaledb_internal.test_status(201);
\set ON_ERROR_STOP 0
SELECT _timescaledb_internal.test_status(304);
SELECT _timescaledb_internal.test_status(400);
SELECT _timescaledb_internal.test_status(401);
SELECT _timescaledb_internal.test_status(404);
SELECT _timescaledb_internal.test_status(500);
SELECT _timescaledb_internal.test_status(503);
\set ON_ERROR_STOP 1

-- This function runs the test 5 times, because each time the internal C function is choosing a random length to send from the server on each socket read. We hit many cases this way.
CREATE OR REPLACE FUNCTION mocker(TEXT) RETURNS SETOF TEXT AS
$BODY$
DECLARE
r TEXT;
BEGIN
FOR i in 1..5 LOOP
SELECT _timescaledb_internal.test_status_mock($1) INTO r;
RETURN NEXT r;
END LOOP;
RETURN;
END
$BODY$
LANGUAGE 'plpgsql';

select * from mocker(
        E'HTTP/1.1 200 OK\r\n'
        'Content-Type: application/json; charset=utf-8\r\n'
        'Date: Thu, 12 Jul 2018 18:33:04 GMT\r\n'
        'ETag: W/\"e-upYEWCL+q6R/++2nWHz5b76hBgo\"\r\n'
        'Server: nginx\r\n'
        'Vary: Accept-Encoding\r\n'
        'Content-Length: 14\r\n'
        'Connection: Close\r\n\r\n'
        '{\"status\":200}');
select * from mocker(
        E'HTTP/1.1 201 OK\r\n'
        'Content-Type: application/json; charset=utf-8\r\n'
        'Vary: Accept-Encoding\r\n'
        'Content-Length: 14\r\n'
        'Connection: Close\r\n\r\n'
        '{\"status\":201}');

\set ON_ERROR_STOP 0
\set test_string 'HTTP/1.1 404 Not Found\r\nContent-Length: 14\r\nConnection: Close\r\n\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
\set test_string 'Content-Length: 14\r\nConnection: Close\r\n\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
\set test_string 'HTTP/1.1 404 Not Found\r\nContent-Length: 14\r\nConnection: Close\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
SELECT _timescaledb_internal.test_status_mock(:'test_string');
\set ON_ERROR_STOP 1

-- Test parsing version response
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"status": "200", "current_timescaledb_version": "10.1.0"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "9.2.0"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "9.1.2"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0"}');

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc1"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc2"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc1"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-alpha"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "123456789"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "!@#$%"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": ""}');

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": " 10 "}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "a"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "a.b.c"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1a"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1+rc1"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1.1"}');
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"}');
SET timescaledb.telemetry_level=basic;
-- Connect to a bogus host and path to test error handling in telemetry_main()
SELECT _timescaledb_internal.test_telemetry_main_conn('noservice.timescale.com', 'path');

-- Test error message and override for when telemetry is disabled
SET timescaledb.telemetry_level=off;
SELECT get_telemetry_report();
SELECT get_telemetry_report(NULL);
SELECT * FROM json_object_keys(get_telemetry_report(always_display_report := true)::json) AS key
WHERE key != 'os_name_pretty';

-- Test telemetry report contents
SET timescaledb.telemetry_level=basic;

SELECT * FROM json_object_keys(get_telemetry_report()::json) AS key
WHERE key != 'os_name_pretty';

-- check telemetry picks up flagged content from metadata
SELECT json_object_field(get_telemetry_report()::json,'db_metadata');

-- check timescaledb_telemetry.cloud
SELECT json_object_field(get_telemetry_report()::json,'instance_metadata');

SELECT json_object_field(get_telemetry_report()::json,'num_hypertables');

SELECT json_object_field(get_telemetry_report()::json,'num_continuous_aggs');

CREATE TABLE device_readings (
      observation_time  TIMESTAMPTZ       NOT NULL
);
SELECT table_name FROM create_hypertable('device_readings', 'observation_time');

SELECT json_object_field(get_telemetry_report()::json,'num_hypertables');
