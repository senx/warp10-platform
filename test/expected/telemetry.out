-- This file and its contents are licensed under the Apache License 2.0.
-- Please see the included NOTICE for copyright information and
-- LICENSE-APACHE for a copy of the license.
\c :TEST_DBNAME :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status(int) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status_ssl(int) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status_ssl' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_status_mock(text) RETURNS JSONB
    AS :MODULE_PATHNAME, 'ts_test_status_mock' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_validate_server_version(response text)
    RETURNS TEXT
    AS :MODULE_PATHNAME, 'ts_test_validate_server_version' LANGUAGE C IMMUTABLE PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_telemetry_main_conn(text, text)
RETURNS BOOLEAN AS :MODULE_PATHNAME, 'ts_test_telemetry_main_conn' LANGUAGE C IMMUTABLE PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_telemetry(host text = NULL, servname text = NULL, port int = NULL) RETURNS JSONB AS :MODULE_PATHNAME, 'ts_test_telemetry' LANGUAGE C IMMUTABLE PARALLEL SAFE;
INSERT INTO _timescaledb_catalog.metadata VALUES ('foo','bar',TRUE);
INSERT INTO _timescaledb_catalog.metadata VALUES ('bar','baz',FALSE);
\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
SELECT _timescaledb_internal.test_status_ssl(200);
 test_status_ssl 
-----------------
 {"status": 200}
(1 row)

SELECT _timescaledb_internal.test_status_ssl(201);
 test_status_ssl 
-----------------
 {"status": 201}
(1 row)

\set ON_ERROR_STOP 0
SELECT _timescaledb_internal.test_status_ssl(304);
ERROR:  endpoint sent back unexpected HTTP status: 304
SELECT _timescaledb_internal.test_status_ssl(400);
ERROR:  endpoint sent back unexpected HTTP status: 400
SELECT _timescaledb_internal.test_status_ssl(401);
ERROR:  endpoint sent back unexpected HTTP status: 401
SELECT _timescaledb_internal.test_status_ssl(404);
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status_ssl(500);
ERROR:  endpoint sent back unexpected HTTP status: 500
SELECT _timescaledb_internal.test_status_ssl(503);
ERROR:  endpoint sent back unexpected HTTP status: 503
\set ON_ERROR_STOP 1
SELECT _timescaledb_internal.test_status(200);
   test_status   
-----------------
 {"status": 200}
(1 row)

SELECT _timescaledb_internal.test_status(201);
   test_status   
-----------------
 {"status": 201}
(1 row)

\set ON_ERROR_STOP 0
SELECT _timescaledb_internal.test_status(304);
ERROR:  endpoint sent back unexpected HTTP status: 304
SELECT _timescaledb_internal.test_status(400);
ERROR:  endpoint sent back unexpected HTTP status: 400
SELECT _timescaledb_internal.test_status(401);
ERROR:  endpoint sent back unexpected HTTP status: 401
SELECT _timescaledb_internal.test_status(404);
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status(500);
ERROR:  endpoint sent back unexpected HTTP status: 500
SELECT _timescaledb_internal.test_status(503);
ERROR:  endpoint sent back unexpected HTTP status: 503
\set ON_ERROR_STOP 1
-- This function runs the test 5 times, because each time the internal C function is choosing a random length to send from the server on each socket read. We hit many cases this way.
CREATE OR REPLACE FUNCTION mocker(TEXT) RETURNS SETOF TEXT AS
$BODY$
DECLARE
r TEXT;
BEGIN
FOR i in 1..5 LOOP
SELECT _timescaledb_internal.test_status_mock($1) INTO r;
RETURN NEXT r;
END LOOP;
RETURN;
END
$BODY$
LANGUAGE 'plpgsql';
select * from mocker(
        E'HTTP/1.1 200 OK\r\n'
        'Content-Type: application/json; charset=utf-8\r\n'
        'Date: Thu, 12 Jul 2018 18:33:04 GMT\r\n'
        'ETag: W/\"e-upYEWCL+q6R/++2nWHz5b76hBgo\"\r\n'
        'Server: nginx\r\n'
        'Vary: Accept-Encoding\r\n'
        'Content-Length: 14\r\n'
        'Connection: Close\r\n\r\n'
        '{\"status\":200}');
     mocker      
-----------------
 {"status": 200}
 {"status": 200}
 {"status": 200}
 {"status": 200}
 {"status": 200}
(5 rows)

select * from mocker(
        E'HTTP/1.1 201 OK\r\n'
        'Content-Type: application/json; charset=utf-8\r\n'
        'Vary: Accept-Encoding\r\n'
        'Content-Length: 14\r\n'
        'Connection: Close\r\n\r\n'
        '{\"status\":201}');
     mocker      
-----------------
 {"status": 201}
 {"status": 201}
 {"status": 201}
 {"status": 201}
 {"status": 201}
(5 rows)

\set ON_ERROR_STOP 0
\set test_string 'HTTP/1.1 404 Not Found\r\nContent-Length: 14\r\nConnection: Close\r\n\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  endpoint sent back unexpected HTTP status: 404
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  endpoint sent back unexpected HTTP status: 404
\set test_string 'Content-Length: 14\r\nConnection: Close\r\n\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
\set test_string 'HTTP/1.1 404 Not Found\r\nContent-Length: 14\r\nConnection: Close\r\n{\"status\":404}';
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
SELECT _timescaledb_internal.test_status_mock(:'test_string');
ERROR:  could not parse HTTP response
\set ON_ERROR_STOP 1
-- Test parsing version response
SELECT * FROM _timescaledb_internal.test_validate_server_version('{"status": "200", "current_timescaledb_version": "10.1.0"}');
 test_validate_server_version 
------------------------------
 10.1.0
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1"}');
 test_validate_server_version 
------------------------------
 10.1
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10"}');
 test_validate_server_version 
------------------------------
 10
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "9.2.0"}');
 test_validate_server_version 
------------------------------
 9.2.0
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "9.1.2"}');
 test_validate_server_version 
------------------------------
 9.1.2
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0"}');
 test_validate_server_version 
------------------------------
 1.0.0
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc1"}');
 test_validate_server_version 
------------------------------
 1.0.0-rc1
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc2"}');
 test_validate_server_version 
------------------------------
 1.0.0-rc2
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-rc1"}');
 test_validate_server_version 
------------------------------
 1.0.0-rc1
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-alpha"}');
 test_validate_server_version 
------------------------------
 1.0.0-alpha
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "123456789"}');
 test_validate_server_version 
------------------------------
 123456789
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "!@#$%"}');
 test_validate_server_version 
------------------------------
 
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": ""}');
 test_validate_server_version 
------------------------------
 
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": " 10 "}');
 test_validate_server_version 
------------------------------
 
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "a"}');
 test_validate_server_version 
------------------------------
 a
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "a.b.c"}');
 test_validate_server_version 
------------------------------
 a.b.c
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1a"}');
 test_validate_server_version 
------------------------------
 10.1.1a
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1+rc1"}');
 test_validate_server_version 
------------------------------
 
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "10.1.1.1"}');
 test_validate_server_version 
------------------------------
 10.1.1.1
(1 row)

SELECT * FROM _timescaledb_internal.test_validate_server_version('{"current_timescaledb_version": "1.0.0-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"}');
 test_validate_server_version 
------------------------------
 
(1 row)

SET timescaledb.telemetry_level=basic;
-- Connect to a bogus host and path to test error handling in telemetry_main()
SELECT _timescaledb_internal.test_telemetry_main_conn('noservice.timescale.com', 'path');
WARNING:  telemetry could not connect to "noservice.timescale.com"
 test_telemetry_main_conn 
--------------------------
 f
(1 row)

-- Test error message and override for when telemetry is disabled
SET timescaledb.telemetry_level=off;
SELECT get_telemetry_report();
INFO:  Telemetry is disabled. Call get_telemetry_report(always_display_report := true) to view the report locally.
 get_telemetry_report 
----------------------
 
(1 row)

SELECT get_telemetry_report(NULL);
INFO:  Telemetry is disabled. Call get_telemetry_report(always_display_report := true) to view the report locally.
 get_telemetry_report 
----------------------
 
(1 row)

SELECT * FROM json_object_keys(get_telemetry_report(always_display_report := true)::json) AS key
WHERE key != 'os_name_pretty';
             key             
-----------------------------
 db_uuid
 license
 os_name
 os_release
 os_version
 data_volume
 db_metadata
 build_os_name
 install_method
 installed_time
 last_tuned_time
 num_hypertables
 build_os_version
 exported_db_uuid
 instance_metadata
 build_architecture
 last_tuned_version
 postgresql_version
 related_extensions
 num_continuous_aggs
 timescaledb_version
 compressed_heap_size
 num_reorder_policies
 compressed_index_size
 compressed_toast_size
 uncompressed_heap_size
 uncompressed_index_size
 uncompressed_toast_size
 num_drop_chunks_policies
 num_compressed_hypertables
 build_architecture_bit_size
(31 rows)

-- Test telemetry report contents
SET timescaledb.telemetry_level=basic;
SELECT * FROM json_object_keys(get_telemetry_report()::json) AS key
WHERE key != 'os_name_pretty';
             key             
-----------------------------
 db_uuid
 license
 os_name
 os_release
 os_version
 data_volume
 db_metadata
 build_os_name
 install_method
 installed_time
 last_tuned_time
 num_hypertables
 build_os_version
 exported_db_uuid
 instance_metadata
 build_architecture
 last_tuned_version
 postgresql_version
 related_extensions
 num_continuous_aggs
 timescaledb_version
 compressed_heap_size
 num_reorder_policies
 compressed_index_size
 compressed_toast_size
 uncompressed_heap_size
 uncompressed_index_size
 uncompressed_toast_size
 num_drop_chunks_policies
 num_compressed_hypertables
 build_architecture_bit_size
(31 rows)

-- check telemetry picks up flagged content from metadata
SELECT json_object_field(get_telemetry_report()::json,'db_metadata');
 json_object_field 
-------------------
 {"foo": "bar"}
(1 row)

-- check timescaledb_telemetry.cloud
SELECT json_object_field(get_telemetry_report()::json,'instance_metadata');
 json_object_field 
-------------------
 {"cloud": "ci"}
(1 row)

SELECT json_object_field(get_telemetry_report()::json,'num_hypertables');
 json_object_field 
-------------------
 "0"
(1 row)

SELECT json_object_field(get_telemetry_report()::json,'num_continuous_aggs');
 json_object_field 
-------------------
 "0"
(1 row)

CREATE TABLE device_readings (
      observation_time  TIMESTAMPTZ       NOT NULL
);
SELECT table_name FROM create_hypertable('device_readings', 'observation_time');
   table_name    
-----------------
 device_readings
(1 row)

SELECT json_object_field(get_telemetry_report()::json,'num_hypertables');
 json_object_field 
-------------------
 "1"
(1 row)

